momento-sdk/src/main/java/momento/sdk/retry/DefaultRetryEligibilityStrategy.java

```diff
@@ -0,0 +1,58 @@
+package momento.sdk.retry;
+
+import io.grpc.Status;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * DefaultRetryEligibilityStrategy is an implementation of the RetryEligibilityStrategy interface
+ * that determines whether a gRPC call is eligible for retry based on its method name and status
+ * code.
+ */
+public class DefaultRetryEligibilityStrategy implements RetryEligibilityStrategy {
+
+  private static final Set<Status.Code> RETRYABLE_STATUS_CODES = new HashSet<>();
+
+  static {
+    RETRYABLE_STATUS_CODES.add(Status.Code.UNAVAILABLE);
+    RETRYABLE_STATUS_CODES.add(Status.Code.INTERNAL);
+  }
+
+  private static final Set<String> RETRYABLE_GRPC_FULL_METHOD_NAMES =
+      new HashSet<String>() {
```

Comment:
We should have the sorted set methods represented here, unless none of them are retryable.